src/btree/bt_handle.c

```diff
@@ -120,7 +121,8 @@ __wt_btree_open(WT_SESSION_IMPL *session, const char *op_cfg[])
 		WT_ERR_MSG(session, EINVAL, "expected a 'file:' URI");
 
 	WT_ERR(__wt_block_manager_open(session, filename, dhandle->cfg,
-	    forced_salvage, readonly, btree->allocsize, &btree->bm));
+	    forced_salvage, F_ISSET(btree, WT_BTREE_READONLY),
+	    btree->allocsize, &btree->bm));
```

Comment:
Do the `__wt_block_manager_open` and `bt->checkpoint_load` calls still need the readonly parameter? Could they instead check the btree structure?

Reply:
Currently btree structure is not referenced in block manager. so I think better to have readonly parameter.